syntax = "proto2";

package depot;

// Enum representing the type of message being sent.
enum ServerMessageType {
    INIT = 0;
    CONFIG = 1;
    REPORT = 2;
}

// Message representing the type of the top-level message
message TypeSignifier {
    required ServerMessageType type = 1;
}

// Message sent by worker sever to register with depot config dispatcher.
message ServerInit {
    required string name = 1;
    required string ip = 2;
}

// Message sent by depot config dispatcher to server with new config to process.
message ServerConfig {
    required string uuid = 1;
    required string name = 2;
    required string body = 3;
}

// Message sent by worker server to depot dispatcher as heartbeat.
message ServerReport {
    required string server_uuid = 1;
    required uint32 ep_num = 3;

    required bool has_config = 4;
    optional bool done = 5;
    optional string config_uuid = 6;
}
